reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -53,7 +55,14 @@ final class StreamSubscriber[F[_]: ConcurrentEffect, A](val sub: StreamSubscribe
 
 object StreamSubscriber {
   def apply[F[_]: ConcurrentEffect, A]: F[StreamSubscriber[F, A]] =
-    fsm[F, A].map(new StreamSubscriber(_))
+    Queue
+      .bounded[F, Either[Throwable, Option[A]]](1)
+      .flatMap(fsm[F, A](_, 1).map(new StreamSubscriber(_)))
+
+  def apply[F[_]: ConcurrentEffect, A](batchSize: Int): F[StreamSubscriber[F, A]] =
```

Comment:
I think `maxDemand` or `outstandingDemand` are more appropriate names for the `batchSize` argument. While this value is an upper bound on `Chunk` size, it isn't a lower bound and I think "batch" seems to evoke fixed size chunks, which we aren't producing.